### PR TITLE
GH-677 Add home admin feature.

### PR DESCRIPTION
**Pull request by vLuckyyy**
## Summary by CodeRabbit

- **New Features**
	- Introduced a command handler for managing user homes, allowing admins to set, delete, teleport to, and list homes.
	- Added a method to check if a specific home exists for a player.
	- Enhanced translation support for home management actions, including administrative messages.
	- Added a new record for encapsulating player and home associations.
	- Implemented a resolver for handling player home entries in command contexts.

- **Bug Fixes**
	- Ensured backward compatibility with existing home-checking methods.

- **Documentation**
	- Expanded documentation for new methods and functionalities related to home management.

### CODE REVIEW

**Review by igoyek — COMMENTED** [review]
obraz
obraz
obraz

I found some bugs @vLuckyyy

**Comment by vLuckyyy** [comment]
@coderabbitai full review

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/HomeAdminCommand.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                .notice(translate -> translate.home().homeList())
+                .placeholder("{HOMES}", homes)
+                .player(sender.getUniqueId())
+                .send();
+
+            return;
+        }
+
+        this.homeManager.deleteHome(uniqueId, home.getName());
+    }
```
[the remark this answers is not in the document]
Comment (Rollczi) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/HomeAdminCommand.java:87]: rel

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    ) {
+        Viewer viewer = this.viewerService.any(invocation.sender());
+
+        NoticeBroadcast offlinePlayer = this.noticeService.create()
+            .notice(translation -> translation.argument().offlinePlayer())
+            .viewer(viewer);
+
+        if (!rawInput.hasNext()) {
+            return ParseResult.failure(offlinePlayer);
+        }
```
[the remark this answers is not in the document]
Comment (Rollczi) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java:67]: rel

File: eternalcore-plugin/build.gradle.kts [hunk 3]
```diff
@@ -27,8 +27,8 @@ eternalShadowCompiler {
             "META-INF/**",
         )
 
-        dependsOn(":eternalcore-core:test")
-        dependsOn(":eternalcore-core:checkstyleMain")
+//        dependsOn(":eternalcore-core:test")
+//        dependsOn(":eternalcore-core:checkstyleMain")
```
Comment (Rollczi) [inline eternalcore-plugin/build.gradle.kts:31]: sigma

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            return SuggestionResult.of(this.homeManager.getHomes(player.getUniqueId()).stream()
+                    .map(Home::getName)
+                    .collect(Collectors.toList()))
+                .appendLeft(playerName);
+        }
+
+        return SuggestionResult.empty();
+    }
+
+    private NoticeBroadcast homeNotice(Collection<Home> homesCount, Player player, UUID uniqueId) {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java:137]: homesCount -> homes

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                .player(uniqueId);
+
+            return ParseResult.failure(home);
+        }
+
+        String homeName = rawInput.next();
+        Optional<Home> home = this.homeManager.getHome(uniqueId, homeName);
+
+        if (home.isEmpty()) {
+            Collection<Home> homesCount = this.homeManager.getHomes(uniqueId);
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java:90]: homesCount -> homes

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.eternalcode.core.feature.home.homeadmin;
+
+import com.eternalcode.core.feature.home.Home;
+import org.bukkit.entity.Player;
+
+public record PlayerHomeEntry(Player player, Home home) {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java:6]: public???

**Review by Rollczi — CHANGES_REQUESTED** [review]
The bot gave good suggestions, fix these messages, etc

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        String homes = homesCount.stream()
+            .map(h -> h.getName())
+            .collect(Collectors.joining(HOME_DELIMITER));
+
+        return this.noticeService.create()
+            .notice(translate -> translate.home().homeList())
+            .placeholder(HOME_LIST_PLACEHOLDER_PREFIX, homes)
+            .placeholder(PLAYER_NAME_PLACEHOLDER_PREFIX, player.getName())
+            .player(uniqueId);
+    }
```
[the remark this answers is not in the document]
Comment (vLuckyyy) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeMultiArgument.java:158]: i dont want hardcoded "none"

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/HomeManager.java [hunk 8]
```diff
@@ -123,6 +123,17 @@ public boolean hasHome(UUID playerUniqueId, String name) {
         return homes.containsKey(name);
     }
 
+    @Override
+    public boolean hasHome(UUID playerUniqueId, Home home) {
+        Map<String, Home> homes = this.userHomes.get(playerUniqueId);
```
Comment (P1otrulla) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/HomeManager.java:128]: we can use getOrDefault here and we eliminate if checking i guess

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package com.eternalcode.core.feature.home.homeadmin;
+
+import com.eternalcode.core.feature.home.Home;
+import org.bukkit.entity.Player;
+
+record PlayerHomeEntry(Player player, Home home) {
```
Comment (P1otrulla) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java:6]: that can be memory leak from dependencies in Player structure? Maybe im wrong

**Review by P1otrulla — APPROVED** [review]
Good job, but PlayerHomeArgument will be working when player run cmd: /home? as defult home?

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java (hunk 9 above)
Comment (Rollczi) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/homeadmin/PlayerHomeEntry.java:6]: I don't think so

**Review by Jakubk15 — APPROVED** [review]
@coderabbitai carried the review as always

**Comment by CitralFlo** [comment]
LGTM

**Review by CitralFlo — APPROVED** [review]
LGTM

**Review by imDMK — APPROVED** [review]

**Review by Rollczi — APPROVED** [review]

**Pull request merged**